### ISSUE TITLE
Reuse LambdaParser in REPL autocompletion

### ISSUE THREAD

**Issue by rexim** [morganey-lang/Morganey#89]
Derived from https://github.com/rexim/Morganey/pull/86#discussion_r70655203

**Comment by keddelzz** [comment]
The most obvious solution: We could extract all regex (as strings) from the Parser into a separate object. Both parser and ReplAutocompletion could use it from the object. This would allow us [this](https://github.com/rexim/Morganey/blob/master/src/main/scala/me/rexim/morganey/ReplAutocompletion.scala#L142) regex.

**Comment by rexim** [comment]
And for modules autocomplete I literally did this:

``` diff
diff --git a/src/main/scala/me/rexim/morganey/ReplAutocompletion.scala b/src/main/scala/me/rexim/morganey/ReplAutocompletion.scala
index bd329d8..e8d42cb 100644
--- a/src/main/scala/me/rexim/morganey/ReplAutocompletion.scala
+++ b/src/main/scala/me/rexim/morganey/ReplAutocompletion.scala
@@ -4,8 +4,9 @@ import java.io.File
 import java.util.{List => Jlist}

 import jline.console.completer.Completer
-import me.rexim.morganey.ast.MorganeyBinding
+import me.rexim.morganey.ast.{MorganeyBinding, MorganeyLoading}
 import me.rexim.morganey.module.ModuleFinder
+import me.rexim.morganey.syntax.LambdaParser

 import scala.annotation.tailrec
 import scala.util.Try
@@ -92,13 +93,16 @@ class ReplAutocompletion(globalContext: () => List[MorganeyBinding],
   }

   private object LoadStatement {
-    private val loadStatement = "load (.*)".r
-
-    def unapply(line: String): Option[(List[String], Boolean)] =
-      Option(line) collect {
-        case loadStatement(modulePath) =>
-          (modulePath.split("\\.").toList, modulePath.endsWith("."))
+    def unapply(line: String): Option[(List[String], Boolean)] = {
+      val result = LambdaParser.parse(LambdaParser.loading, line)
+
+      if (result.successful) {
+        val modulePath = result.get.modulePath
+        Some((modulePath.split("\\.").toList, modulePath.endsWith(".")))
+      } else {
+        None
       }
+    }
   }

   private def matchingDefinitions(line: String, knownVariableNames: List[String]): List[String] =
```

and it sorta worked. The only downside was I couldn't see any available variants when I didn't type at least one symbol of a module path. The `loadStatement` regex just allows to not have a name for the module while `LambdaParser.loading` does not.

Ideally, that would be cool to just reuse our existing parsers. But it's quite hard right now. Maybe later we will come with something. :)

**Comment by keddelzz** [comment]
I would like to do a trial-implementation of this issue. @rexim Do you have a problem with me creating a PR as a design-idea once this is ready?

**Comment by rexim** [comment]
@keddelzz sure! It's a great idea! I actually like to use GitHub's PR mechanism as a form of communication. :)

**Comment by keddelzz** [comment]
closed with merge of PR #104

**Comment by rexim** [comment]
@keddelzz thanks for closing that! My focus is scattered accross so many things these day I even forget to close the issues. :D

**Comment by keddelzz** [comment]
No problem. :)

**Comment by keddelzz** [comment]
I have, after all, forgotten to put `closes #89` into the description of the PR.
